<a href="https://colab.research.google.com/github/dpetrovic89/Telenor-Churn/blob/main/1%20EDA_Exploratory_Data_Analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Import Libraries & Load Data

In [ ]:
import pandas as pd

In [ ]:
df_description = pd.read_csv('https://raw.githubusercontent.com/dpetrovic89/data/main/Variable-Explanation-Test-Data-Scientist_2022.csv', header=None, )
df_description.head(15)

0                                                  1   2
0          SUBSCRIBER_ID                                   ID of subscriber NaN
1               BNUM_OUT   Number of different numbers who subscriber calls NaN
2                BNUM_IN  Number of different numbers that make calls to... NaN
3                  CHURN               Churn event, 1 - churn, 0 non-churn. NaN
4         PROD_CNT_MACRO  Number of bundles that customer activated (ave... NaN
5            REV_BUN_MAC  Revenue from bundles that customer activated (... NaN
6                REV_OUT           Total Revenue (average in last 3 months) NaN
7      USAGE_OUT_INT_DUR  Number of outgoing international minutes (aver... NaN
8   USAGE_OUT_OFFNET_DUR  Number of outgoing minutes to competitor netwo... NaN
9    USAGE_OUT_ONNET_DUR  Number of outgoing minutes in the network (ave... NaN
10            LNE_TENURE                              User tenure in months NaN
11             TOPUP_AMT  Total topup (refill) amount (average in last 3... NaN
12             TOPUP_CNT  Number of topups (refill) (average in last 3 m... NaN
13               Segment                                  Customer segment. NaN

In [ ]:
df= pd.read_csv('https://raw.githubusercontent.com/dpetrovic89/data/main/Data-Churn-Test-Data-Scientist_2021.csv' )
df.head()

SUBSCRIBER_ID  BNUM_OUT  BNUM_IN  ...  TOPUP_AMT  TOPUP_CNT    Segment
0              1       2.0      3.0  ...        NaN        NaN  Segment 1
1              2      52.0     32.0  ...        2.0        3.0  Segment 2
2              3       NaN      NaN  ...        NaN        0.0  Segment 3
3              4       4.0      5.0  ...        4.0        2.0  Segment 1
4              5       0.0     14.0  ...        6.0        2.0  Segment 5

[5 rows x 14 columns]

# EDA

In [ ]:
df.isna().sum()

SUBSCRIBER_ID              0
BNUM_OUT                 488
BNUM_IN                  488
CHURN                      0
PROD_CNT_MACRO          2825
REV_BUN_MAC             3977
REV_OUT                  508
USAGE_OUT_INT_DUR        401
USAGE_OUT_OFFNET_DUR     401
USAGE_OUT_ONNET_DUR      401
LNE_TENURE                 0
TOPUP_AMT               2263
TOPUP_CNT               1909
Segment                    2
dtype: int64

**What to do with missing values?**

SUBSCRIBER_ID              0

BNUM_OUT                 488          Mean 

BNUM_IN                  488          Mean 

CHURN                      0

PROD_CNT_MACRO          2825          0

REV_BUN_MAC             3977          0

REV_OUT                  508          Mean

USAGE_OUT_INT_DUR        401          0

USAGE_OUT_OFFNET_DUR     401          0

USAGE_OUT_ONNET_DUR      401          0

LNE_TENURE                 0

TOPUP_AMT               2263          0

TOPUP_CNT               1909          0

Segment                    2          ▶ Data Leak --- Drop Column


In [ ]:
df_description.head(15)

0                                                  1   2
0          SUBSCRIBER_ID                                   ID of subscriber NaN
1               BNUM_OUT   Number of different numbers who subscriber calls NaN
2                BNUM_IN  Number of different numbers that make calls to... NaN
3                  CHURN               Churn event, 1 - churn, 0 non-churn. NaN
4         PROD_CNT_MACRO  Number of bundles that customer activated (ave... NaN
5            REV_BUN_MAC  Revenue from bundles that customer activated (... NaN
6                REV_OUT           Total Revenue (average in last 3 months) NaN
7      USAGE_OUT_INT_DUR  Number of outgoing international minutes (aver... NaN
8   USAGE_OUT_OFFNET_DUR  Number of outgoing minutes to competitor netwo... NaN
9    USAGE_OUT_ONNET_DUR  Number of outgoing minutes in the network (ave... NaN
10            LNE_TENURE                              User tenure in months NaN
11             TOPUP_AMT  Total topup (refill) amount (average in last 3... NaN
12             TOPUP_CNT  Number of topups (refill) (average in last 3 m... NaN
13               Segment                                  Customer segment. NaN

In [ ]:
df["CHURN"].value_counts(normalize=True)


0    0.7826
1    0.2174
Name: CHURN, dtype: float64

In [ ]:
What are average values of numerical features for churned users?

df[df["CHURN"] == 1].mean()


Object `users` not found.


SUBSCRIBER_ID           2603.317387
BNUM_OUT                   4.309198
BNUM_IN                    5.663405
CHURN                      1.000000
PROD_CNT_MACRO             0.451852
REV_BUN_MAC                1.973684
REV_OUT                    1.505060
USAGE_OUT_INT_DUR          0.660534
USAGE_OUT_OFFNET_DUR       4.574057
USAGE_OUT_ONNET_DUR       21.101904
LNE_TENURE               117.922723
TOPUP_AMT                  6.321534
TOPUP_CNT                  1.416856
dtype: float64

In [ ]:
df.groupby(["CHURN"]).describe(percentiles=[])

SUBSCRIBER_ID                                 ... TOPUP_CNT                
              count         mean          std  min  ...       std  min  50%   max
CHURN                                               ...                          
0            3913.0  2471.938155  1441.037342  2.0  ...  2.751919  0.0  2.0  31.0
1            1087.0  2603.317387  1448.428467  1.0  ...  1.819927  0.0  1.0  18.0

[2 rows x 72 columns]